lib/web/plug/params_validator.ex

```diff
@@ -0,0 +1,142 @@
+# Copyright(c) 2015-2024 ACCESS CO., LTD. All rights reserved.
+
+use Croma
+
+defmodule Antikythera.Plug.ParamsValidator do
+  @moduledoc """
+  Plug to validate request parameters.
+
+  ## Usage
+
+  First, generate parameter validation models using `Antikythera.ParamStringStruct` or `Antikythera.BodyJsonStruct`.
+
+      defmodule MyQueryParams do
+        use Antikythera.ParamStringStruct,
+          fields: [
+            item_id: Croma.PosInteger
+          ]
+      end
+
+  Then, you can install this plug to validate parameters, as shown below.
+
+      plug Antikythera.Plug.ParamsValidator, :validate, query_params: MyQueryParams
+
+  In the controller function, you can access the validated parameters as follows.
+
+      defun index(conn :: v[Conn.t()]) :: Conn.t() do
+        %MyQueryParams{item_id: item_id} = conn.assigns.validated.query_params
+
+        # Use item_id
+      end
+
+  ## Supported parameter types
+
+  - `:path_matches`: Path parameters
+  - `:query_params`: Query parameters
+  - `:body`: Request body (JSON)
+      - Note: The module must be generated by `Antikythera.BodyJsonStruct`.
+  - `:headers`: Request headers
+      - Note: The keys must be lowercase. (e.g., `:"x-item-id"` instead of `:"X-Item-Id"`)
+  - `:cookies`: Cookies
+
+  Multiple parameter types can be validated at once.
+
+      plug Antikythera.Plug.ParamsValidator, :validate, path_matches: MyPathParams, body: MyBody
+  """
```

Comment:
It would be better to write about invalid case like `Call parameter_validation_error/3`

Reply:
Added explanation about error handling at 4140383.